### ISSUE TITLE
[REQUEST] ECR image hook

### ISSUE THREAD

**Issue by ITProKyle** [onicagroup/runway#490]
With the release of public ECR (in response to Docker Hub limits) and the ability to use ECR images as Lambda function deployment package ECR is now a very appealing option. There is a void in capabilities to build, test, and push images.

## Acceptance Criteria

- [x] build docker image
- [x] ~optionally test the image~ leaving this out for now, until there is a need for it
- [x] push the image that was built
- [x] push image that has already been built
- [ ] delete image locally and/or from ECR
- [x] provide access to sane docker options in the build/push process
- [ ] return information relevant to use as Lambda function deployment package and with other services

**Comment by ITProKyle** [comment]
After starting work on the hook, I noticed most of the logic I was writing was around docker and not ECR. So, this is now going to be implemented as a docker hook with features to simplify use with ECR.

**Comment by ITProKyle** [comment]
Looking into returning relevant info to use as a Lambda Function deployment package I noticed that CloudFormation documentation does not reflect supporting this yet while the release blog post stated that CloudFormation does support it.

Looking through the API docs, I was able to piece together a working CloudFormation template for testing. I also started a PR to add support to Troposphere (https://github.com/cloudtools/troposphere/pull/1846).
